src/main/scala/chisel3/util/Math.scala

```diff
@@ -99,3 +98,34 @@ object isPow2 {
   def apply(in: BigInt): Boolean = in > 0 && ((in & (in-1)) == 0)
   def apply(in: Int): Boolean = apply(BigInt(in))
 }
+
+
+object unsignedBitLength {
+  /** Return the number of bits required to encode a specific value, assuming no sign bit is required.
+    *
+    * Basically, `n.bitLength` unless n is 0 (since Java believes 0.bitLength == 0), in which case the result is `1`.
+    * @param in - the number to be encoded.
+    * @return - an Int representing the number of bits to encode.
+    */
+  def apply(in: BigInt): Int = {
+    require(in >= 0)
+    if (in == 0) {
+      // 0.bitLength returns 0 - thank you Java.
+      1
```

Comment:
Was this the behavior we wanted? (I'm sure we brought this up, I don't remember the resolution and I don't think it was written down)

Reply:
I believe so, but we can review at next dev meeting.

Reply:
Return 0, not 1.